addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -2260,14 +2262,16 @@ export class OdooEditor extends EventTarget {
                 // We Collapse the selection and bypass deleteRange
                 // if the range content is only one ZWS.
                 sel.collapseToStart();
-                if (BACKSPACE_ONLY_COMMANDS.includes(method)) {
+                if (isBackSpaceCommand) {
                     this._applyRawCommand(method);
                 }
                 return;
             }
-            this.deleteRange(sel);
-            if (BACKSPACE_ONLY_COMMANDS.includes(method)) {
-                return true;
+            if (isBackSpaceCommand || hasTableFullySelected(range) || !closestElement(sel.anchorNode, 'td.o_selected_td')) {
```

Comment:
```suggestion
            if (
                BACKSPACE_ONLY_COMMANDS.includes(method) ||
                isTableFullySelected(range) ||
                closestElement(sel.anchorNode, 'td.o_selected_td')
            ) {
```
Please also add a comment, in particular to explain what you explained to me regarding the third condition.

Reply:
Here's what you explained to me regarding the third condition:
![image](https://github.com/odoo/odoo/assets/24205914/6a6a314a-72c0-4937-abe2-17bf8efd3167)
I'd like, at the minimum, that explanation (nicely phrased for the reader of course) to be included in the comment.